### PR TITLE
6 core get article comments

### PR DESCRIPTION
**Pull request by MScott1118**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -68,3 +68,55 @@ describe("GET /api/article/:article_id", () => {
...
+          expect(typeof article.title).toBe("string");
+          expect(typeof article.topic).toBe("string");
+          expect(typeof article.author).toBe("string");
+          expect(typeof article.created_at).toBe("string");
+          expect(typeof article.votes).toBe("number");
+          expect(typeof article.article_img_url).toBe("string");
+          expect(typeof article.article_id).toBe("number");
+          expect(typeof article.comment_count).toBe("string");
+        });
+        expect(count).toBe(36);
```
Comment (roisinoreilly) [inline __tests__/app.test.js]: see my comments on your last PR re: checking the length of the array, using .objectContaining or .toMatchObject and removing lines that are checking an array is an array if you're then iterating over it

File: __tests__/app.test.js [hunk 2]
```diff
@@ -68,3 +68,55 @@ describe("GET /api/article/:article_id", () => {
...
+    return request(app).get("/api/articles/1/comments").expect(200);
+  });
+  test("should return an array of all comment objects with correct keys", () => {
+    return request(app)
+      .get("/api/articles/1/comments")
+      .expect(200)
+      .then((res) => {
+        expect(Array.isArray(res.body.comments)).toBe(true);
+        res.body.comments.forEach((comment) => {
+          expect(typeof comment.comment_id).toBe("number");
```
Comment (roisinoreilly) [inline __tests__/app.test.js:111]: you know this is going to be article id 1 so test that it is

**Review by roisinoreilly — CHANGES_REQUESTED** [review]
good work so far, I've left comments on your previous PR for changes I'd like you to make to your tests. I'd also like

- [ ] a test for if an article has no comments associated with it (I think you can use article id 2 for this)
- [ ] a 400 test for an invalid article id eg. /articles/banana/comments
- [ ] when you're testing specific articles' comments actually test for that id, not just expecting any number

once you've done that send us another PR :)

File: __tests__/app.test.js [hunk 3]
```diff
@@ -68,3 +68,99 @@ describe("GET /api/article/:article_id", () => {
...
+      .expect(201);
+  });
+  test("should return the new comment object with the correct keys", () => {
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send({
+        username: "jessjelly",
+        body: "Great article!",
+      })
+      .expect(201)
```
Comment (Harrison365) [inline __tests__/app.test.js:147]: No need to repeat expect(201) check

File: __tests__/app.test.js [hunk 4]
```diff
@@ -68,3 +68,99 @@ describe("GET /api/article/:article_id", () => {
...
+      .then((res) => {
+        expect(Array.isArray(res.body.comment)).toBe(true);
+        expect(typeof res.body.comment[0].comment_id).toBe("number");
+        expect(typeof res.body.comment[0].votes).toBe("number");
+        expect(typeof res.body.comment[0].created_at).toBe("string");
+        expect(typeof res.body.comment[0].author).toBe("string");
+        expect(typeof res.body.comment[0].body).toBe("string");
+        expect(typeof res.body.comment[0].article_id).toBe("number");
+      });
```
Comment (Harrison365) [inline __tests__/app.test.js:156]: We can check the actual article_id here : )

File: db/models.js [hunk 5]
```diff
@@ -23,3 +23,48 @@ exports.fetchArticleByID = (article_id) => {
...
+
+exports.insertComment = (article_id, newComment) => {
+  const body = newComment.body;
+  const username = newComment.username;
+  return db.query(
+    `INSERT INTO comments (body, author, article_id)
+  VALUES ($1, $2, $3)
+  RETURNING *`,
+    [body, username, article_id]
+  );
```
Comment (Harrison365) [inline db/models.js:69]: Nice 👍

**Comment by Harrison365** [comment]
Very Good :) 

Please add a test for the error message on your sad path tests (i.e. not just the status code)

Please also add the following tests:
400: Invalid article id e.g. banana 🍌
404: username not found (one not in the user data)
400: missing field (e.g. no body property on request)

**Pull request merged**